posthog/test/test_should_update_person_prop.py

```diff
@@ -0,0 +1,157 @@
+# 10| set_once | yes           | after                      | set_once    | no
+
+# Refers to migration 0173_should_update_person_props_function
+# This is a Postgres function we use in the plugin server
+class TestShouldUpdatePersonProp(BaseTest):
+
+    # tests cases 1 and 2 from the table
+    def test_update_non_existent_prop(self):
+        person = Person.objects.create(
+            team=self.team, properties={}, properties_last_updated_at={}, properties_last_operation={}
+        )
+
+        with connection.cursor() as cursor:
+            cursor.execute(
+                f"""
+                SELECT 
+                    should_update_person_prop({person.id}, 'non-existent prop', now()::text, 'set'),
+                    should_update_person_prop({person.id}, 'non-existent prop', now()::text, 'set_once')
+            """
+            )
+
+            result = cursor.fetchall()
+            set_op_result = result[0][0]
+            set_once_op_result = result[0][1]
+
+            self.assertEqual(set_op_result, True)
+            self.assertEqual(set_once_op_result, True)
+
+    # tests cases 3 and 4 from the table
+    def test_set_operation_with_earlier_timestamp(self):
+        person = Person.objects.create(
+            team=self.team,
+            properties={"a": 0, "b": 0},
+            properties_last_updated_at={
+                "a": datetime(2050, 1, 1, 1, 1, 1).isoformat(),
+                "b": datetime(2050, 1, 1, 1, 1, 1).isoformat(),
+            },
+            properties_last_operation={"a": "set", "b": "set_once"},
+        )
+
+        with connection.cursor() as cursor:
+            cursor.execute(
+                f"""
+                SELECT 
+                    should_update_person_prop({person.id}, 'a', now()::text, 'set'),
+                    should_update_person_prop({person.id}, 'b', now()::text, 'set')
+            """
+            )
+
+            result = cursor.fetchall()
+            previous_op_set_result = result[0][0]
+            previous_op_set_once_result = result[0][1]
+
+            self.assertEqual(previous_op_set_result, False)
+            self.assertEqual(previous_op_set_once_result, True)
+
+    # tests cases 5 and 6 from the table
+    def test_set_operation_with_older_timestamp(self):
+        person = Person.objects.create(
+            team=self.team,
+            properties={"a": 0, "b": 0},
+            properties_last_updated_at={
+                "a": datetime(2000, 1, 1, 1, 1, 1).isoformat(),
+                "b": datetime(2000, 1, 1, 1, 1, 1).isoformat(),
+            },
+            properties_last_operation={"a": "set", "b": "set_once"},
+        )
+
+        with connection.cursor() as cursor:
+            cursor.execute(
+                f"""
+                SELECT 
+                    should_update_person_prop({person.id}, 'a', now()::text, 'set'),
+                    should_update_person_prop({person.id}, 'b', now()::text, 'set')
```

Comment:
same here add the set_once ops here from 9 & 10.

Reply:
Same comment as above